### PR TITLE
split code into modules and simplify

### PR DESCRIPTION
**Pull request by yyyc514**

### CODE REVIEW

**Comment by yyyc514** [comment]
Lots of little things here, but the individual PRs should be clear.  The commits that break things out into modules are just moving code around (and perhaps renaming the functions), not changing anything else.

File: src/highlight.js [hunk 1]
```diff
@@ -3,34 +3,19 @@ Syntax highlighting with language autodetection.
...
-    globalObject.hljs = factory({});
-
-    // Finally register the global hljs with AMD.
-    if(typeof define === 'function' && define.amd) {
-      define([], function() {
-        return globalObject.hljs;
-      });
-    }
-  }
+import deepFreeze from './vendor/deep_freeze';
```
Comment (egor-rogov) [inline src/highlight.js]: What's the reason to put it into `vendor` (instead of `utils`)?

Comment (yyyc514) [reply src/highlight.js]: It's pretty much borrowed code from elsewhere, hence vendoring it.

Comment (egor-rogov) [reply src/highlight.js]: I don't insist, although for me a comment pointing to the original repo would be enough.

Are you done with the PR or you're going to add some more changes?

Comment (yyyc514) [reply src/highlight.js]: It's a Rails habit, but I like it.  Doesn't hurt anything to organize it nicely.

> Are you done with the PR or you're going to add some more changes?

Done.  I try to not assign things or label them WIP if they aren't finished.

Comment (yyyc514) [reply src/highlight.js]: This finally lays the groundwork for a parse tree, but that's another PR.

**Comment by yakov116** [comment]
@yyyc514 your awesome! 
Quick question, some places your using var, some your using const. Is there a reason for that?

**Comment by taufik-nurrohman** [comment]
@yakov116 migration requires adaptation and `var` is ~~back~~forward compatible. It should be removed in the future for module system but for day-to-day basis, `var` is perfectly okay for web browsers.

**Comment by yyyc514** [comment]
@yakov116 It's a legacy codebase.  Over time we'll switch to let/const as I notice it.  It's not a high priority to do it all at once.

**Pull request merged**